### PR TITLE
[MRG+1] Clean up unit examples

### PR DESCRIPTION
**Pull request by dstansby**
Adds in all the comments from  #8254

### CODE REVIEW

File: examples/units/artist_tests.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
 ============
 
-Test unit support with each of the matplotlib primitive artist types
+Test unit support with each of the Matplotlib primitive artist types.
 
-The axes handles unit conversions and the artists keep a pointer to
-their axes parent, so you must init the artists with the axes instance
-if you want to initialize them with unit data, or else they will not
+The axis handles unit conversions and the artists keep a pointer to
+their axis parent, so you must initialize the artists with the axis instance
```
Comment (QuLogic) [inline examples/units/artist_tests.py]: I'm thinking this is still a bit long; can you break the sentence in two: "their axis parent. You must initialize ..."?

File: examples/units/bar_demo2.py [hunk 2]
```diff
@@ -3,12 +3,12 @@
 Bar demo with units
 ===================
 
-plot using a variety of cm vs inches conversions.  The example shows
-how default unit instrospection works (ax1), how various keywords can
+A plot using a variety of centimetre vs inches conversions.  This example shows
```
Comment (QuLogic) [inline examples/units/bar_demo2.py]: vs inches -> and inch

Is this line not too long? Or is everything else just wrapped a bit short?

**Comment by vollbier** [comment]
Perhaps it would be helpful to add a comment  that the module `basic_units` resides in the `mpl_examples/units/`  directory (I tripped over that some time ago and was dumbfound for a while searching through conda and pip).

**Comment by dstansby** [comment]
Backported to `2.0.x` via. https://github.com/matplotlib/matplotlib/commit/2fb0da8d28c5c8054e01ffca2344ac1cab2ef5b8

**Pull request merged**